pkg/apis/istio/v1alpha1/istioinstaller_types.proto

```diff
@@ -0,0 +1,597 @@
+one or more components, which correspond to Istio components (Pods) in the cluster.
+
+k8s API: this API is a pass through to k8s resource settings for Istio k8s resources. It allows customizing Istio k8s
+resources like Affinity, Resource requests/limits, PodDisruptionBudgetSpec, Selectors etc. in a more consistent and
+k8s specific way compared to values.yaml. See KubernetesResourcesSpec in this file for details.
+
+2. values.yaml
+
+The entirety of values.yaml settings is accessible through InstallSpec (see CommonComponentSpec/ValueOverrides).
+This API will gradually be deprecated and values there will be moved either into CRDs that are used to directly
+configure components or, in the case of k8s settings, will be replaced by the new API above.
+
+3. k8s resource overlays
+
+Once a manifest is rendered from InstallSpec, a further specialization can be applied by specifying k8s resource
+overlays. The concept is similar to kustomize, where JSON patches are applied for object paths. This allows
+customization at the lowest level and eliminates the need to create ad-hoc template parameters, or edit templates.
+
+EXAMPLES
+
+1. Default Istio install
+
+spec: ~
+
+2. Default minimal profile install
+
+spec:
+  baseSpecPath: istio-minimal.yaml
+
+3. Default install with telemetry disabled
+
+spec:
+  telemetry:
+    enabled: false
+
+4. Default install with each feature installed to different namespace and security components in separate namespaces
+
+spec:
+  traffic_management:
+    namespace: istio-traffic-management
+  policy:
+    namespace: istio-policy
+  telemetry:
+    namespace: istio-telemetry
+  config_management:
+    namespace: istio-config-management
+  security:
+    citadel:
+      namespace: istio-citadel
+    cert_manager:
+      namespace: istio-cert-manager
+    node_agent:
+      namespace: istio-node-agent
+
+5. Default install with specialized k8s settings for pilot
+
+spec:
+  traffic_management:
+    pilot:
+        k8s:
+          resources:
+            limits:
+              cpu: 444m
+              memory: 333Mi
+            requests:
+              cpu: 222m
+              memory: 111Mi
+          readinessProbe:
+            failureThreshold: 44
+            initialDelaySeconds: 11
+            periodSeconds: 22
+            successThreshold: 33
+
+6. Default install with values.yaml customizations for proxy
+
+spec:
+  traffic_management:
+    proxy:
+      valuesOverrides:
+      - global.proxy.enableCoreDump: true
+      - global.proxy.dnsRefreshRate: 10s
+
+7. Default install with modification to container flag in galley
+
+spec:
+  configuration_management:
+    galley:
+      k8s:
+        overlays:
+        - apiVersion: extensions/v1beta1
+          kind: Deployment
+          name: istio-galley
+          patches:
+          - path: spec.template.spec.containers.[name:galley].command.[--livenessProbeInterval]
+            value: --livenessProbeInterval=123s
+*/
+
+
+message IstioInstall {
```

Comment:
I strongly believe we should stop using "install" in any of the new CRDs and in operator. 

Install is not the primary purpose - but operation, i.e. continuous configuration of the components after install.
Yes, it can also install ( or remove ) components as needed - or they may be pre-installed by vendor. 

It is highly confusing to use the term install - unless there is something that can never be changed after install ( and I hope we don't have anything in this category ).

Reply:
I'm not attached to any particular name, do you have anything in mind?
IstioInstallation maybe? An installation implies longer term continuity. 
IstioOperatorConfig?

Reply:
I agree with @costinm.  The operator should be providing automating the operation of Istio.  The type I created, as part of the skeleton, is named `IstioControlPlane`, because the operator is managing the lifecycle of a control plane.  'IstioOperatorConfig` I think is inappropriate because you are not configuring the operator.

Simply put, an operator is a controller that watches specific resources and acts on them, performing tasks that a human operator would do.  A simple of example of this, although not called an operator, is the k8s `Deployment`.  It's controller watches those resources, acting on them as changes are made, e.g. rolling out a new set of pods, increasing replica count, etc.

Reply:
Sure, sgtm. I changed it to IstioControlPlane.